core/spec/lib/devise/model/session_limitable_spec.rb

```diff
@@ -0,0 +1,13 @@
+require 'rails_helper'
+
+RSpec.describe Devise::Models::SessionLimitable do
```

Comment:
@ouranos I've tried many options, including: 
```ruby
RSpec.describe Devise::Models::SessionLimitable do
  include Warden::Test::Helpers
  
  subject { user.reload }

  let(:user) { build(:user, sso_session: 'session_id') }

  before do
    login_as(user, run_callbacks: true)
```

Nothing seems to be triggering Warden's callbacks... Do you have any clue?

Reply:
See https://github.com/maestrano/mno-enterprise/commit/432b629956266b85c5c3f90543622dff96ce463e for some working specs.

I had to:
- reload the user class after updating the settings so that the module is loaded like in [user_spec](https://github.com/maestrano/mno-enterprise/blob/master/core/spec/models/mno_enterprise/user_spec.rb#L5-L11)
- do a request spec so we go through the full rails stack

The request spec is in `api` to have a controller endpoint to request.

If we use `sso_session` we'll most likely won't need `#update_unique_session_id!` (and the associated hook) as we update sso_session in Maestrano-hub
Just need some testing to make sure it’s always properly updated and to store it in the warden scope.

Reply:
thanks!